src/instance.jl

```diff
@@ -89,16 +93,18 @@ function MOI.delete!(m::AbstractInstance, vr::MOI.VariableReference)
     for cr in rm
         MOI.delete!(m, cr)
     end
+    splice!(m.varrefs, vf)
     if haskey(m.varnames, vr.value)
         delete!(m.namesvar, m.varnames[vr.value])
         delete!(m.varnames, vr.value)
     end
 end
 
 MOI.isvalid(m::AbstractInstance, cr::MOI.ConstraintReference) = !iszero(m.constrmap[cr.value])
+MOI.isvalid(m::AbstractInstance, vr::MOI.VariableReference) = in(vf, m.varrefs)
```

Comment:
Wouldn't this be extremely inefficient if there are many variables ? I think we should rather use a `Set` than a `Vector`

Reply:
Probably. But it's not clear if that will make much of a difference in the grand scheme of things. Depends how often people plan on checking whether a VariableReference is valid.